org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/JavaLanguageServerPlugin.java

```diff
@@ -174,6 +177,27 @@ public void start(BundleContext bundleContext) throws Exception {
 		if (System.getProperty(AssistOptions.PROPERTY_SubstringMatch) == null) {
 			System.setProperty(AssistOptions.PROPERTY_SubstringMatch, "false");
 		}
+		// https://github.com/redhat-developer/vscode-java/issues/3797
+		Job initializeAfterLoad = new Job("Initialize After Load") {
```

Comment:
Does this actually fix the issue ? I ask because when this job is scheduled, the client has probably not yet connected, nor has the server finished the `initialized` request. Looking though `JavaCore.initializeAfterLoad(..)`, it seems like the kind of thing that should run after project import ? Does that make sense or is this location correct.

https://github.com/eclipse-jdt/eclipse.jdt.core/blob/55d99cde490111cad4874bc05eec03397ca5bc0e/org.eclipse.jdt.core/model/org/eclipse/jdt/core/JavaCore.java#L4702-L4727

Reply:
Yes, it does. I have tried without it and Java LS always rebuilds indexes at https://github.com/eclipse-jdtls/eclipse.jdt.ls/pull/3296/commits/cb2bbb73d6822d54bf1fd304f8463cc180c183c3#diff-f23d0258ce24117c110acd6753e8490a09fbdf0bc039d662bdffe279c8c28469R742
We don't want to rebuild indexes every time.
You can try to comment it, and you will get `rebuild indexes` in the log.